src/CommonLib/LDAPUtils.cs

```diff
@@ -370,8 +374,16 @@ public IEnumerable<string> DoRangedRetrieval(string distinguishedName, string at
             {
                 SearchResponse response;
 
-                response = (SearchResponse) conn.SendRequest(searchRequest);
-
+                try
+                {
+                    _searchQuerySemaphore.WaitOne();
```

Comment:
I have some minor concerns around waits like this. This could result in a lock up in the daemon that can only be resolved by user intervention (a restart). There are other `WaitOne(...)` methods that support setting a timeout. We may want to consider utilization of one of those while consuming the boolean return value to gate further use of the connection.

This introduces additional error handing concerns but at the benefit of not falling into deadlock.

Reply:
Maybe on boolean `false` throw an exception or something else that will get caught and interpreted at a layer above.